File: src/main/java/me/elsiff/morefish/condition/MCMMOSkillCondition.java
package me.elsiff.morefish.condition;

import me.elsiff.morefish.MoreFish;
import me.elsiff.morefish.hooker.MCMMOHooker;
import org.bukkit.entity.Player;

/**
 * Created by elsiff on 2017-06-20.
 */
public class MCMMOSkillCondition implements Condition {
    private String skillType;
    private int level;

    public MCMMOSkillCondition(String skillType, int level) {
        this.skillType = skillType;
        this.level = level;
    }

    @Override
    public boolean isSatisfying(Player player) {
        MCMMOHooker hooker = MoreFish.getInstance().getMCMMOHooker();

        if (hooker == null) {
            MoreFish.getInstance().getLogger().severe("MCMMO Not Found!");
            return false;
        }

        return (hooker.getSkillLevel(player, skillType) >= level);
    }
}

File: src/main/kotlin/me/elsiff/morefish/fishing/condition/McmmoSkillCondition.kt
package me.elsiff.morefish.fishing.condition

import me.elsiff.morefish.fishing.competition.FishingCompetition
import me.elsiff.morefish.hooker.McmmoHooker
import me.elsiff.morefish.hooker.PluginHooker
import org.bukkit.entity.Item
import org.bukkit.entity.Player

/**
 * Created by elsiff on 2019-01-12.
 */
class McmmoSkillCondition(
    private val mcmmoHooker: McmmoHooker,
    private val skillType: String,
    private val minLevel: Int
) : FishCondition {
    override fun check(caught: Item, fisher: Player, competition: FishingCompetition): Boolean {
        PluginHooker.checkEnabled(mcmmoHooker, fisher.server.pluginManager)
        return mcmmoHooker.skillLevelOf(fisher, skillType) >= minLevel
    }
}
